src/cairoUtilFuncGen/memory/memoryRead.ts

```diff
@@ -102,23 +108,13 @@ export class MemoryReadGen extends StringIndexedFuncGen {
 
   private getOrCreate(typeToRead: CairoType): GeneratedFunctionInfo {
     const funcName = `WM${this.generatedFunctionsDef.size}_READ_${typeToRead.typeName}`;
-    const resultCairoType = typeToRead.toString();
-    const [reads, pack] = serialiseReads(typeToRead, readFelt, readFelt);
     const funcInfo: GeneratedFunctionInfo = {
       name: funcName,
-      code: [
-        `func ${funcName}{range_check_ptr, warp_memory : DictAccess*}(loc: felt) ->(val: ${resultCairoType}){`,
-        `    alloc_locals;`,
-        ...reads.map((s) => `    ${s}`),
-        `    return (${pack},);`,
-        '}',
-      ].join('\n'),
-      functionsCalled: [this.requireImport(...DICT_READ)],
+      code: endent`fn ${funcName}(loc: felt) -> ${typeToRead.toString()}{
```

Comment:
I think the `#[implicits(warp_memory: WarpMemory)]` is missing here

Reply:
True